dotnet/src/Plugins/Plugins.Memory/TextSearchPlugin.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.ComponentModel;
+using System.Linq;
+using System.Text.Encodings.Web;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.Search;
+
+namespace Microsoft.SemanticKernel.Plugins;
+
+/// <summary>
+/// Text search plugin
+/// </summary>
+/// <remarks>
+/// Initializes a new instance of the <see cref="TextSearchPlugin{T}"/> class.
+/// </remarks>
+/// <param name="service">The text search service instance to use.</param>
+public sealed class TextSearchPlugin<T>(ITextSearchService<T> service) where T : class
+{
+    private readonly ITextSearchService<T> _service = service;
+
+    /// <summary>
+    /// The usage of JavaScriptEncoder.UnsafeRelaxedJsonEscaping here is considered safe in this context
+    /// because the JSON result is not used for any security sensitive operations like HTML injection.
+    /// </summary>
+    private static readonly JsonSerializerOptions s_jsonOptionsCache = new()
+    {
+        Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping,
+    };
+
+    /// <summary>
+    /// Perform a search for content related to the specified query.
+    /// </summary>
+    /// <param name="query">The text to search for.</param>
+    /// <param name="count">The number of results to return. Default is 1.</param>
+    /// <param name="offset">The number of results to skip. Default is 0.</param>
+    /// <param name="cancellationToken">A cancellation token to observe while waiting for the task to complete.</param>
+    /// <returns>A task that represents the asynchronous operation. The value of the TResult parameter contains the search results as a string.</returns>
+    [KernelFunction, Description("Perform a search for content related to the specified query.")]
+    public async Task<string> SearchAsync(
```

Comment:
Can we return `List<T>` or `IReadOnlyList<T>` instead of `string`?

Reply:
Weslie, as proposed an approach that eliminates the need for this plugin, I'm going to try that next